File: clean_css.py
import os
import re

css_dir = 'd:/work/nishikant/elcon-network/elcon-frontend/src'

# Regex to match these CSS blocks and anything inside them
patterns = [
    r'\.table-footer\s*\{[^}]*\}',
    r'\.total-entries\s*\{[^}]*\}',
    r'\.pagination\s*\{[^}]*\}',
    r'\.page-btn\s*\{[^}]*\}',
    r'\.page-btn:hover\s*\{[^}]*\}',
    r'\.page-btn:hover:not\(:disabled\)\s*\{[^}]*\}',
    r'\.page-btn\.active-page\s*\{[^}]*\}',
    r'\.page-btn:disabled\s*\{[^}]*\}',
    r'\/\*\s*Footer and Pagination\s*\*\/'
]

for root, dirs, files in os.walk(css_dir):
    for file in files:
        if file.endswith('.css') and file != 'index.css':
            file_path = os.path.join(root, file)
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            new_content = content
            for p in patterns:
                new_content = re.sub(p, '', new_content, flags=re.MULTILINE)
            
            # Remove any extra blank lines left over
            new_content = re.sub(r'\n{3,}', '\n\n', new_content)
            
            if new_content != content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f'Cleaned up {file_path}')

print('Done cleaning CSS files.')

File: clean_css.js
const fs = require('fs');
const path = require('path');

const cssDir = 'd:/work/nishikant/elcon-network/elcon-frontend/src';

const patterns = [
    /\.table-footer\s*\{[^}]*\}/g,
    /\.total-entries\s*\{[^}]*\}/g,
    /\.pagination\s*\{[^}]*\}/g,
    /\.page-btn\s*\{[^}]*\}/g,
    /\.page-btn:hover\s*\{[^}]*\}/g,
    /\.page-btn:hover:not\(:disabled\)\s*\{[^}]*\}/g,
    /\.page-btn\.active-page\s*\{[^}]*\}/g,
    /\.page-btn\.active\s*\{[^}]*\}/g,
    /\.page-btn:disabled\s*\{[^}]*\}/g,
    /\/\*\s*Footer and Pagination\s*\*\//g
];

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            results.push(file);
        }
    });
    return results;
}

const files = walk(cssDir);

files.forEach(file => {
    if (file.endsWith('.css') && !file.endsWith('index.css')) {
        let content = fs.readFileSync(file, 'utf8');
        let newContent = content;
        
        patterns.forEach(p => {
            newContent = newContent.replace(p, '');
        });
        
        newContent = newContent.replace(/\n{3,}/g, '\n\n');
        
        if (newContent !== content) {
            fs.writeFileSync(file, newContent, 'utf8');
            console.log('Cleaned up ' + file);
        }
    }
});

console.log('Done cleaning CSS files.');
